### PR TITLE
Add simple explanation messages to some errors

### PR DESCRIPTION
**Pull request by qarmin**
This helps to find what exactly is wrong with bad code.

example from https://github.com/godotengine/godot/issues/50471#issuecomment-880581824

Before
```
ERROR: Condition "!method" is true. Returning: false
   at: is_builtin_method_vararg (core/variant/variant_call.cpp:1125)
```

After
```
ERROR: Cannot find "call_me" method in "Nil" class.
   at: is_builtin_method_vararg (core/variant/variant_call.cpp:1161)
```

### CODE REVIEW

File: core/variant/callable.cpp [hunk 1]
```diff
@@ -386,13 +386,13 @@ Error Signal::connect(const Callable &p_callable, const Vector<Variant> &p_binds
...
 	Object *object = get_object();
-	ERR_FAIL_COND(!object);
+	ERR_FAIL_COND_MSG(!object, "Object was removed before disconnection.");
 	object->disconnect(name, p_callable);
 }
 
 bool Signal::is_connected(const Callable &p_callable) const {
 	Object *object = get_object();
-	ERR_FAIL_COND_V(!object, false);
+	ERR_FAIL_COND_V_MSG(!object, false, "Object was removed before.");
```
Comment (Ansraer) [inline core/variant/callable.cpp:396]: Removed before what?

File: core/variant/variant_utility.cpp [hunk 2]
```diff
@@ -1121,7 +1121,7 @@ static void register_utility_function(const String &p_name, const Vector<String>
 		name = name.substr(1, name.length() - 1);
 	}
 	StringName sname = name;
-	ERR_FAIL_COND(utility_function_table.has(sname));
+	ERR_FAIL_COND_MSG(utility_function_table.has(sname), "Function \"" + name + "\" already was registered as utility function.");
```
Comment (Ansraer) [inline core/variant/variant_utility.cpp:1124]: Imo this should be "[...] was already [...]"

File: drivers/vulkan/vulkan_context.cpp [hunk 3]
```diff
@@ -2099,16 +2099,11 @@ void VulkanContext::local_device_push_command_buffers(RID p_local_device, const
...
-	if (err == VK_ERROR_OUT_OF_DEVICE_MEMORY) {
-		print_line("Vulkan: Out of device memory!");
-	}
-	if (err == VK_ERROR_DEVICE_LOST) {
-		print_line("Vulkan: Device lost!");
-	}
-	ERR_FAIL_COND(err);
+
+	ERR_FAIL_COND_MSG(err == VK_ERROR_OUT_OF_HOST_MEMORY, "Vulkan: Out of host memory!");
+	ERR_FAIL_COND_MSG(err == VK_ERROR_OUT_OF_DEVICE_MEMORY, "Vulkan: Out of device memor!");
```
Comment (Ansraer) [inline drivers/vulkan/vulkan_context.cpp:2104]: memory

File: modules/navigation/godot_navigation_server.cpp [hunk 4]
```diff
@@ -496,7 +496,7 @@ COMMAND_1(free, RID, p_object) {
 		agent_owner.free(p_object);
 
 	} else {
-		ERR_FAIL_COND("Invalid ID.");
+		ERR_FAIL_MSG("Trying to free invalid ID: " + itos(p_object.get_id()));
```
Comment (Ansraer) [inline modules/navigation/godot_navigation_server.cpp:499]: All the pther messages end with a dot.

File: scene/3d/skeleton_3d.cpp [hunk 5]
```diff
@@ -491,7 +491,7 @@ void Skeleton3D::set_bone_name(int p_bone, const String &p_name) {
 
 	for (int i = 0; i < bone_size; i++) {
 		if (i != p_bone) {
-			ERR_FAIL_COND(bones[i].name == p_name);
+			ERR_FAIL_COND_MSG(bones[i].name == p_name, "Bone \"" + p_name + "\" already exists in Skeleton3D.");
```
Comment (Ansraer) [inline scene/3d/skeleton_3d.cpp:494]: Previously you used "sheleton", now "Skeleton3D". It would probably be better to use the same term.

**Review by Ansraer — CHANGES_REQUESTED** [review]
Just some small typos I spotted.

**Review by EricEzaM — COMMENTED** [review]
I wrote these reviews a few days ago but forgot to submit.

**Pull request closed without merging**